MobiusCore/Source/AsyncDispatchQueueConnectable.swift

```diff
@@ -47,21 +38,14 @@ final class AsyncDispatchQueueConnectable<Input, Output>: Connectable {
         self.init(AnyConnectable(underlyingConnectable), acceptQueue: acceptQueue)
     }
 
-    func connect(_ consumer: @escaping (Output) -> Void) -> Connection<Input> {
-        let disposalStatus = Synchronized(value: DisposalStatus.notDisposed)
+    func connect(_ consumer: @escaping Consumer<Output>) -> Connection<Input> {
+        let protectedConsumer = Synchronized<Optional<Consumer<Output>>>(value: consumer)
 
         let connection = underlyingConnectable.connect { value in
-            // Don’t forward if we’re currently waiting to dispose the connection.
-            //
-            // NOTE: the underlying consumer must be called inside the critical region accessing disposalStatus, or we
-            // could potentially enter the .pendingDispose state before or during the consumer call. This means that the
-            // underlying consumer must not call our connection’s acceptClosure or disposeClosure, or it will deadlock.
-            //
-            // This is OK in our existing use case because the underlying consumer is always flipEventsToLoopQueue from
-            // MobiusController’s initializer, which enters the actual Mobius loop asynchronously. If we exposed this
-            // class, it would be a scary edge case.
-            disposalStatus.read { status in
-                guard status != .pendingDispose else { return }
+            protectedConsumer.read { consumer in
+                guard let consumer = consumer else {
+                    MobiusHooks.errorHandler("cannot consume value after dispose", #file, #line)
```

Comment:
The previous code had an issue with retaining the consumer reference, which would lead to a somewhat nebulous crash when its unowned loop reference was accessed by emitting an event after disposal. This code is functionally equivalent, but the cause of the error should be clearer now.